### PR TITLE
Add Travis CI Azure build

### PR DESCRIPTION
**Pull request by evaldofelipe**
This PR add a travis automated process to grant the full tests coverage.
The steps tested here are:

- Build docker image
- Provisioning all Azure environment
- Build image with Packer
- Destroy test environment

The only artifact that continues here is the template image.

### CODE REVIEW

**Review by OctavioBR — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+sudo: required
+language: python
+services:
+  - docker
+before_install:
+  - sudo apt-get -qq update
+  - sudo apt-get install -o Dpkg::Options::="--force-confold" --force-yes -y docker-ce
+  - mkdir keys
+  - ssh-keygen -b 2048 -t rsa -f /home/travis/build/NeowayLabs/packer-images/keys/id_rsa -q -N ""
+  - cp hack/test.sh .
```
Comment (lborguetti) [inline .travis.yml]: Why we need this?

File: Makefile [hunk 2]
```diff
@@ -101,6 +105,11 @@ packer-fmt: ##@packer Rewrite configuration files to a canonical format and styl
 	packer-images \
 	bash packer-fmt.sh
 
+.PHONY: test
+test: ##@test Run CI tests on travis to all build steps
+	@echo "Starting CI tests..."
+	bash test.sh
```
Comment (lborguetti) [inline Makefile]: So, maybe this approach is better and we don't need copy the test.sh ;)

```suggestion
	bash hack/test.sh
```

**Review by lborguetti — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (evaldofelipe) [reply Makefile]: I've tried use this approach, but the script was execute the paths inside the `hack/` path. The make targets still running in `/`

**Review by lborguetti — APPROVED** [review]
LGTM!

**Pull request merged**